src/main/java/org/craftercms/studio/impl/v1/service/content/ContentTypeServiceImpl.java

```diff
@@ -84,28 +84,30 @@ public ContentTypeConfigTO getContentTypeForContent(@ValidateStringParam(name =
 
     @Override
     public boolean isUserAllowed(Set<String> userRoles, ContentTypeConfigTO item) {
-        if (item != null) {
-            String name = item.getName();
-            Set<String> allowedRoles = item.getAllowedRoles();
-            logger.debug("Checking allowed roles on " + name + ". user roles: "
-                    + userRoles + ", allowed roles: " + allowedRoles);
+        if (item == null) {
+            logger.debug("No content type config provided for item '{}' to limit user access, " +
+                    "defaulting to permit the user", item.getName());
+            return true;
+        }
 
-            if (allowedRoles == null || allowedRoles.size() == 0) {
-                return true;
-            } else {
-                boolean notAllowed = Collections.disjoint(userRoles, allowedRoles);
-                if (notAllowed) {
-                    logger.debug(name + " is not allowed for the user.");
-                    return false;
-                } else {
-                    return true;
-                }
-            }
-        } else {
-            logger.debug("no content type config provided. returning true for user access to content type checking.");
+        Set<String> allowedRoles = item.getAllowedRoles();
+        logger.trace("Item '{}' allows roles '{}', checking against user roles '{}'",
+                item.getName(), allowedRoles, userRoles);
 
+        if (allowedRoles == null || allowedRoles.size() == 0) {
```

Comment:
Let's consider using CollectionUtils.isEmpty() for these scenarios?

Reply:
Please take note of this and send it as a separate PR after this one is merged.